File: P3/QytetetJava/src/modeloQytetet/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloQytetet;

/**
 *
 * @author marquirj
 */
public class Dado {
    private static final Dado instance = new Dado();
    private int valor;
    int tirar(){
        return 21;
     }
    int getValor(){
        return this.valor;
    }
    private Dado(){}
    public static Dado getInstance() {
        return instance;
    }
}

File: P3/QytetetRuby/lib/dado.rb
# ENCODING: UTF-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require "singleton"
module QytetetRuby
class Dado
  include Singleton
  attr_reader :valor
  def initialize(valor_)
    @valor = valor_
  end
  def tirar
    @valor=rand(1..6)
    return @valor
  end
  def to_s
    "Dado #{@dado}\n"
  end
end
end